crates/noirc_frontend/src/parser/parser.rs

```diff
@@ -1470,27 +1471,74 @@ mod test {
             let message = format!("Failed to parse:\n{}", program);
             let (op_t, errors) = parse_recover(&parser, program);
             for e in errors {
-                if !e.is_warrning() {
-                    panic!("{}", &message);
+                if !e.is_warning() {
+                    panic!("{} with error {}", &message, e);
                 }
             }
             op_t.expect(&message)
         })
     }
 
+    fn parse_all_warnings<P, T>(parser: P, programs: Vec<&str>) -> Vec<CustomDiagnostic>
+    where
+        P: NoirParser<T>,
+        T: std::fmt::Display,
+    {
+        programs
+            .into_iter()
+            .flat_map(|program| match parse_recover(&parser, program) {
+                (None, errors) => errors,
+                (Some(expr), errors) => {
+                    let mut has_errors = false;
+                    for e in &errors {
+                        if e.is_error() {
+                            has_errors = true;
+                        }
+                    }
+                    match has_errors {
+                        true => {
+                            unreachable!(
+                                "Expected this input to pass with warning:\n{}\nYet it successfully failed with error:\n{}",
+                                program, expr
+                            )
+                        }
+                        false => {}
+                    }
+                    errors
+
```

Comment:
```suggestion
                    if errors.iter().any(|error| error.is_error()) {
                        unreachable!(
                            "Expected this input to pass with warning:\n{program}\nYet it successfully failed with error:\n{expr}",
                        )
                    }
                    errors
```
Although we should check for errors in both cases not just if we have `Some(expr)`.

Reply:
correct